code/modules/mob/living/carbon/human/species_types/IPC.dm

```diff
@@ -171,48 +171,54 @@
 	desc = "An internal power cord hooked up to a battery. Useful if you run on electricity. Not so much otherwise."
 	icon = 'icons/obj/power.dmi'
 	icon_state = "wire1"
+	var/charge_sources = list(/obj/machinery/power/apc, /obj/item/stock_parts/cell) // a list of types we can recharge from
 
 /obj/item/apc_powercord/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
-	if(!istype(target, /obj/machinery/power/apc) || !ishuman(user) || !proximity_flag)
+	if(!is_type_in_list(target, charge_sources) || !ishuman(user) || !proximity_flag)
 		return ..()
 	user.changeNext_move(CLICK_CD_MELEE)
-	var/obj/machinery/power/apc/A = target
+	var/obj/item/stock_parts/cell/C
```

Comment:
Can you replace the single letter var with something more descriptive? Like ``charged_cell`` or just straight up ``cell``.
This goes for the powerdraw_loop proc's args too.

Reply:
no